pkg/scanner/scanner.go

```diff
@@ -81,13 +82,37 @@ func (p *scanParser) checkPortsRemoved(host string) {
 	}
 }
 
-type nmapWrapper struct{}
+type nmapWrapper struct {
+	interfaceName string
+}
 
 func (n *nmapWrapper) Run(ipAddresses []string, ctx context.Context) (result *nmap.Run, warnings []string, err error) {
+	if n.interfaceName != "" {
+		return n.runWithDevice(ipAddresses, ctx)
+	}
+	return n.runWithoutDevice(ipAddresses, ctx)
+}
+
+func (n *nmapWrapper) runWithDevice(ipAddresses []string, ctx context.Context) (result *nmap.Run, warnings []string, err error) {
+	scanner, err := nmap.NewScanner(
+		nmap.WithTargets(ipAddresses...),
+		nmap.WithContext(ctx),
+		nmap.WithSkipHostDiscovery(),
+		nmap.WithInterface(n.interfaceName),
+		nmap.WithUnprivileged(),
+	)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to create scanner scanner: %v", err)
```

Comment:
You've also got an extra `scanner` in the log line.

Reply:
I fixed the typo with the refactor of the functions that had these log lines.